### ISSUE TITLE
Ready() provide a synchronous alternative

### ISSUE THREAD

**Issue by gigi81** [poulfoged/Elasticsearch-Inside#8]
Hi,
I just upgraded the nuget package from the alpha2 to 5.0.0 and I found out that there is a new method called Ready() to call before you can actually use elasticsearch.
Anyway, the method is async and my test classes/methods are all synchronous.
I tried all different combinations suggested here to make the method call synchronous:
http://stackoverflow.com/questions/22628087/calling-async-method-synchronously
With no luck.
https://ci.appveyor.com/project/gigi81/elastico/history
Can you please provide a synchronous alternative for the Ready() method?
Many Thanks
Luigi

**Comment by poulfoged** [comment]
Hello, in 5.0.1 I have added a ReadySync() method including a test that returns void:

.ReadySync()

Does that work for you?

Thanks,

Poul

**Comment by gigi81** [comment]
Hi @poulfoged 
Fantastic! I will try it out and let you know how it goes. But I see you had a proper unit test so I don't expect any problem now.
Many Thanks
Luigi

**Comment by gigi81** [comment]
Hi @poulfoged 
I finally managed to try the ReadySync method but with not much luck.
On my local machine it works. Anyway on appveyor the build still stalls forever and I have to kill it.
For example:
https://ci.appveyor.com/project/gigi81/elastico/build/0.5.0.86

I came to the conclusion that the problem is not specifically in your implementation but anyway it has something to do with xunit or appveyor or the combination of the two together.

The underlying problem I think that is still the fact that the method is not really a sync method, but an async method that is called and waited for.

Any idea or suggestion at this point are welcome.
Thanks

**Comment by poulfoged** [comment]
Hm.. ok wierd that it works here:

https://ci.appveyor.com/project/poulfoged/elasticsearch-inside/build/1.0.70

Maybe we need a fully sync path - only problem is HttpClient .. 

I'll reopen and try some things.

**Comment by gigi81** [comment]
Yeah. I noticed that! That's why I feel like it could be xunit and/or the fact that I'm creating the server in the constructor of the unit test class.

**Comment by gigi81** [comment]
I found the root cause of the issue.
The problem is with XUnit and how it handles async calls that can lead to deadlocks if using .Result and friends. This is why the ReadSync() was dead locking for me.
XUnit provides an interface called IAsyncLifetime to implement in a test class. When calling the Ready() async from the InitializeAsync method in the test class, it finally worked for me.
My life would have probably been easier if I just had the sync call as the previous versions of the library, anyway I guess I learned some new things.
